drivers/gpio/gpio_ameba.c

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (c) 2024 Realtek Semiconductor Corp.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT realtek_ameba_gpio
+
+/* Include <soc.h> before <ameba_soc.h> to avoid redefining unlikely() macro */
+#include <soc.h>
+#include <ameba_soc.h>
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+#include <zephyr/irq.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(gpio_ameba, CONFIG_GPIO_LOG_LEVEL);
+
+#define GPIO_PINNAME(PORT, PIN) (((PORT) << 5) | ((PIN) & 0x1F))
+
+struct gpio_ameba_config {
+	/* gpio_driver_config needs to be first */
+	struct gpio_driver_config common;
+	/* port base address */
+	uint32_t base;
+	/* IO port */
+	int port_num;
```

Comment:
Does it represent a `port` or the `number of ports`? It seems to be different from the comment.

```suggestion
	int port; // or pin
```

Reply:
done